plugin-interface/src/main/scala/mesosphere/marathon/plugin/http/HttpRequestHandlerBase.scala

```diff
@@ -0,0 +1,56 @@
+package mesosphere.marathon.plugin.http
+
+import java.io.{ ByteArrayOutputStream, InputStream, OutputStream }
+import java.net.{ URL, URLConnection }
+import java.nio.file.{ Files, Paths }
+
+import scala.annotation.tailrec
+import scala.util.Try
+
+//scalastyle:off magic.number
+abstract class HttpRequestHandlerBase extends HttpRequestHandler {
+
+  protected[this] def serveResource(path: String, response: HttpResponse): Unit = {
+    def transfer(in: InputStream, out: OutputStream) {
```

Comment:
Have you thought about using Streams + transferTo? The code would look cleaner and I think it uses sendFile under-the-hood, so it would also perform better.